File: src/msg_listener.cpp
#include "ros/ros.h"
#include "beginner_tutorials/vision_feedback.h"

void chatterCallback(const beginner_tutorials::vision_feedback::ConstPtr& msg)
{
  ROS_INFO("I heard feedback x = %f", msg->x);
  ROS_INFO("I heard feedback y = %f", msg->y);
  ROS_INFO("I heard feedback theta = %f", msg->theta);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "msg_listener");

  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("feedback", 1000, chatterCallback);

  ros::spin();

  return 0;
}

File: scripts/msg_listener.py
import rospy
from beginner_tutorials.msg import vision_feedback

def callback(data):
    rospy.loginfo('feedback x = %f', data.x)
    rospy.loginfo('feedback y = %f', data.y)
    rospy.loginfo('feedback theta = %f', data.theta)

def listener():

    # In ROS, nodes are uniquely named. If two nodes with the same
    # name are launched, the previous one is kicked off. The
    # anonymous=True flag means that rospy will choose a unique
    # name for our 'listener' node so that multiple listeners can
    # run simultaneously.
    rospy.init_node('msg_listener', anonymous=True)

    rospy.Subscriber('feedback', vision_feedback, callback)

    # spin() simply keeps python from exiting until this node is stopped
    rospy.spin()

if __name__ == '__main__':
    listener()